### PR TITLE
Fix get_subclasses of non ABC

### LINKED ISSUE #334: utils.get_subclasses not further detecting children classes of an instantiable class
## Expected Behavior

get_subclasses should be able to detect all children classes of any class.

## Current Behavior

get_subclasses can not find children further if a class is considered not to be abstract by inspect.isabstract.

## Possible Solution

Expand and add all subclasses to any class, if it ever have children. If the class itself is an abstract class, do not add itself. If it is not an abstract class, add it as well.

## Steps to Reproduce

1. from sparselm.model._base import CVXEstimator
2. get_subclasses(CVXEstimator) will only find "OrdinaryLeastSquares", "Lasso" and "MIQP_L0".

### PR DESCRIPTION
**Pull request by lbluque**
## Summary

* fixes #334
* add tests for fix

## Checklist

- [X] Code is in the [standard Python style](https://www.python.org/dev/peps/pep-0008/).
      Run [pycodestyle](https://pycodestyle.readthedocs.io/en/latest/) and [flake8](http://flake8.pycqa.org/en/latest/)
      on your local machine.
- [X] Docstrings have been added in the [Google docstring format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).
      Run [pydocstyle](http://www.pydocstyle.org/en/2.1.1/index.html) on your code.
- [X] Tests have been added for any new functionality or bug fixes.
- [X] All existing tests pass.

### CODE REVIEW

**Comment by lbluque** [comment]
@qchempku2017 can you have a quick look and see if you can think of any other things to check in the unit test? thanks!

File: tests/test_utils.py [hunk 1]
```diff
@@ -31,7 +33,29 @@ def test_get_subclasses():
     assert all(
         c in get_subclasses(MCKernel).values() for c in [UniformlyRandom, Metropolis]
     )
-    assert all(c in get_subclasses(ThermalKernel).values() for c in [Metropolis])
+    assert Metropolis in get_subclasses(ThermalKernel).values()
```
Comment (qchempku2017) [inline tests/test_utils.py:36]: Why not have a check for all base classes that uses derived_class_factory, for example, basis, bias and usher, etc?

**Review by qchempku2017 — COMMENTED** [review]
Maybe add more tests to other classes that uses derived_class_factory to instatiate? For example, basis, ushers, bias, etc?

**Review by qchempku2017 — APPROVED** [review]
Great! If all these tests pass, we are free to merge!

**Pull request merged**